services/galley/src/Galley/API/Update.hs

```diff
@@ -98,36 +103,62 @@ unblockConv (usr ::: conn ::: cnv) = do
 joinConversation :: UserId ::: ConnId ::: ConvId ::: JSON -> Galley Response
 joinConversation (zusr ::: zcon ::: cnv ::: _) = do
     c <- Data.conversation cnv >>= ifNothing convNotFound
+    when (Data.isConvDeleted c) $ do
+        Data.deleteConversation cnv
+        throwM convNotFound
```

Comment:
Is there an instance of calling `Data.conversation` where we do _not_ want to delete a conversation from cassandra is it's marked as deleted? Just wondering if we could/should instead use something like `lookupAndDeleteIfDeleted` (or something a bit easier to read) or even change the semantics of the lookup?